### PR TITLE
hsmd: Add fields to hsmd_sign_commitment_tx and hsmd_sign_remote_commitment_tx

### PR DESCRIPTION
**Pull request by ksedgwic**
Add to hsmd_sign_remote_commitment_tx:
- commitment_number
- feerate
- htlcs

Add to hsmd_sign_commitment_tx:
- commitment_number

These fields are needed for validating signers.

[Summary of all changes for validating signers](https://gitlab.com/lightning-signer/docs/-/wikis/C-Lightning-Modifications)

### CODE REVIEW

File: channeld/channeld.c [hunk 1]
```diff
@@ -1034,11 +1034,30 @@ static struct bitcoin_signature *calc_commitsigs(const tal_t *ctx,
...
 	struct bitcoin_signature *htlc_sigs;
 
+	/* Collect the htlcs for call to hsmd. */
+	struct simple_htlc **htlcs = tal_arr(tmpctx, struct simple_htlc *, 0);
+	size_t num_entries = tal_count(htlc_map);
+	for (size_t ndx = 0; ndx < num_entries; ++ndx) {
+		struct htlc const *hh = htlc_map[ndx];
+		if (hh) {
+			struct simple_htlc *simple =
+				new_simple_htlc(NULL,
```
Comment (cdecker) [inline channeld/channeld.c:1044]: Please allocate using the `htlcs` array directly, since you're stealing right below it.

**Comment by cdecker** [comment]
Just a minor nit, otherwise LGTM

ACK https://github.com/ElementsProject/lightning/pull/4981/commits/e11f1f82a12cb3f7c4aeaffbe0554cc3a1d909d6

File: channeld/channeld.c (hunk 1 above)
Comment (ksedgwic) [reply channeld/channeld.c]: fixed

**Comment by ksedgwic** [comment]
@cdecker unclear if I own any of the CI fails

**Comment by ksedgwic** [comment]
Pushed trivial change to trigger CI

**Pull request merged**